### ISSUE TITLE
Broken on Node 18

### ISSUE THREAD

**Issue by kerberjg** [orliesaurus/nodemailer-mailgun-transport#121]
Whenever I try to send emails via this module from my application on Node >= 18, I get the following error:

```
(node:23783) ExperimentalWarning: The Fetch API is an experimental feature. This feature could change at any time
(Use `node --trace-warnings ...` to show where the warning was created)

[Error: Bad Request] {
  status: 400,
  details: { message: 'from parameter is missing' }
}
```

Simply changing the Node version to 12 or 14 mitigates this, but it's not something we can use in production as we need the features of Node 18 for our project.

This is not a duplicate of #116, as in this case the email does not send at all, however the fix for both issues would be to simply upgrade the `mailgun.js` dependency. This module is currently depending on v3.3.0, while v8 is available.

**Comment by orliesaurus** [comment]
@kerberjg - with the latest updates merged in and thank you sam-lord this should be fixed!

**Comment by kerberjg** [comment]
Neat! Thanks a lot 😄✨
